main.go

```diff
@@ -6,6 +6,10 @@ import (
 	"github.com/joyent/containerpilot/core"
 
 	log "github.com/Sirupsen/logrus"
+
+	// Import backends so that they initialize
```

Comment:
Seems like they are logically their own package. I feel that `discovery` should be implementation agnostic. Im sort of following the pattern that heka used for plugins.

Also, main.go needs to at least import `discovery` in order for the init chain to work. And then discovery must import `etcd` and `consul` and other backends even though it doesn't care about the implementation classes.

I can see arguments for both, but there doesn't appear to be a clear winner for me, so I'd opt to follow the existing standard.